plugins/vppv2/aclplugin/descriptor/acl.go

```diff
@@ -0,0 +1,397 @@
+			aclIfIndices := d.getOrCacheInterfaces(acl.Interfaces.Ingress, vppACLIndex, L2)
+			err := d.aclHandler.SetMacIPACLToInterface(vppACLIndex, aclIfIndices)
+			if err != nil {
+				return nil, errors.Errorf("failed to set MAC IP ACL %s to interface(s) %v: %v",
+					acl.Name, acl.Interfaces.Ingress, err)
+			}
+		} else {
+			aclIfInIndices := d.getOrCacheInterfaces(acl.Interfaces.Ingress, vppACLIndex, INGRESS)
+			err = d.aclHandler.SetACLToInterfacesAsIngress(vppACLIndex, aclIfInIndices)
+			if err != nil {
+				return nil, errors.Errorf("failed to set IP ACL %s to interface(s) %v as ingress: %v",
+					acl.Name, acl.Interfaces.Ingress, err)
+			}
+			aclIfEgIndices := d.getOrCacheInterfaces(acl.Interfaces.Egress, vppACLIndex, EGRESS)
+			err = d.aclHandler.SetACLToInterfacesAsEgress(vppACLIndex, aclIfEgIndices)
+			if err != nil {
+				return nil, errors.Errorf("failed to set IP ACL %s to interface(s) %v as egress: %v",
+					acl.Name, acl.Interfaces.Ingress, err)
+			}
+		}
+	}*/
+
+	d.log.Infof("ACL %s configured with ID %d", acl.Name, vppACLIndex)
+
+	metadata = &aclidx.AclMetadata{
+		Index: vppACLIndex,
+		L2:    isL2MacIP,
+	}
+	return metadata, nil
+}
+
+// validateRules provided in ACL. Every rule has to contain actions and matches.
+// Current limitation: L2 and L3/4 have to be split to different ACLs and
+// there cannot be L2 rules and L3/4 rules in the same ACL.
+func (c *AclDescriptor) validateRules(aclName string, rules []*acl.Acl_Rule) ([]*acl.Acl_Rule, bool) {
+	var validL3L4Rules []*acl.Acl_Rule
+	var validL2Rules []*acl.Acl_Rule
+
+	for index, rule := range rules {
+		if rule.GetMatch() == nil {
+			c.log.Warnf("invalid ACL %s: rule %d does not contain match", aclName, index)
+			continue
+		}
+		if rule.GetMatch().GetIpRule() != nil {
+			validL3L4Rules = append(validL3L4Rules, rule)
+		}
+		if rule.GetMatch().GetMacipRule() != nil {
+			validL2Rules = append(validL2Rules, rule)
+		}
+	}
+	if len(validL3L4Rules) > 0 && len(validL2Rules) > 0 {
+		c.log.Warnf("ACL %s contains L2 rules and L3/L4 rules as well. This case is not supported, only L3/L4 rules will be resolved",
+			aclName)
+		return validL3L4Rules, false
+	} else if len(validL3L4Rules) > 0 {
+		return validL3L4Rules, false
+	} else {
+		return validL2Rules, true
+	}
+}
+
+func (d *AclDescriptor) Delete(key string, acl *acl.Acl, metadata *aclidx.AclMetadata) error {
+	if metadata.L2 {
+		/*if acl.GetInterfaces() != nil {
+			err := c.aclHandler.RemoveMacIPIngressACLFromInterfaces(vppACLIndex, c.getInterfaces(acl.Interfaces.Ingress))
+			if err != nil {
+				return errors.Errorf("failed to remove MAC IP interfaces from ACL %s: %v",
+					acl.AclName, err)
+			}
+		}*/
+		// Remove ACL L2.
+		err := d.aclHandler.DeleteMacIPACL(metadata.Index)
+		if err != nil {
+			return errors.Errorf("failed to delete MACIP ACL %s: %v", acl.Name, err)
+		}
+		// Unregister.
+		//d.l2AclIndexes.UnregisterName(acl.AclName)
+		d.log.Debugf("MACIP ACL %s deleted", acl.Name)
```

Comment:
To decrease the log and make it more readable, I would recommend not to log that something was created/modified/deleted (what is happening is already printed by scheduler)